lib/grape/validations/params_scope.rb

```diff
@@ -130,13 +165,14 @@ def required?
       # Adds a parameter declaration to our list of validations.
       # @param attrs [Array] (see Grape::DSL::Parameters#requires)
       def push_declared_params(attrs, **opts)
+        opts[:declared_params_scope] ||= self
```

Comment:
Dup this structure, don't modify caller. 

```
opts = opts.merge(decared_params_scope: self) unless opts.key?(:declared_params_scope)
```

Reply:
Yes,I should be more rigorous.